webextension/privileged/ui/cloudstorage.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* global ExtensionAPI */
+
+/**
+ * Handles the Cloud Downloads UI in download panel.
+ */
+
+"use strict";
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+
+this.cloudstorage = class extends ExtensionAPI {
```

Comment:
FWIW, the documentation states this class is instantiated once for every add-on that uses the API ( https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/reference.html#extensionapi-class ). I appreciate this might not matter right now because there's only 1 consumer anyway and the API ships with that 1 add-on, but most of this logic looks to me like it shouldn't live in the class but in a static/global of some sort. Might be best to leave this for now...